scylla/src/routing.rs

```diff
@@ -5,8 +5,38 @@ use std::num::NonZeroU16;
 use thiserror::Error;
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
+
+/// Token is a result of computing a hash of a primary key
+///
+/// It is basically an i64 with one caveat: i64::MIN is not
+/// a valid token. It is used to represent infinity.
+/// For this reason tokens are normalized - i64::MIN
+/// is replaced with i64::MAX. See this fragment of
+/// Scylla code for more information:
+/// <https://github.com/scylladb/scylladb/blob/4be70bfc2bc7f133cab492b4aac7bab9c790a48c/dht/token.hh#L32>
+///
+/// This struct is a wrapper over i64 that performs this normalization
+/// when initialized using `new()` method.
 pub struct Token {
-    pub value: i64,
+    value: i64,
+}
+
+impl Token {
+    /// Creates a new token with given value, normalizing the value if necessary
+    #[inline]
+    pub fn new(value: i64) -> Self {
+        Self {
+            value: if value == i64::MIN { i64::MAX } else { value },
+        }
+    }
+
+    /// Invalid Token - contains i64::MIN as value.
+    pub(crate) const INVALID: Self = Token { value: i64::MIN };
```

Comment:
I suggest mentioning why we even need this.

Reply:
Done